### PR TITLE
Add documentation for parameters

### PR DESCRIPTION
**Pull request by vasyl-purchel**
- adds new annotation method `parameter`
  - adds default `in` option set to `:path` and `:required` for
    parameters specified in the route path
  - adds documentation for parameter description from options:
     parameter name, description: 'parameter description'

### CODE REVIEW

File: lib/doc_my_routes/doc/hash_helpers.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Define error classes
```
Comment (punkle) [inline lib/doc_my_routes/doc/hash_helpers.rb]: I think this comment is misplaced

File: lib/doc_my_routes/doc/hash_helpers.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+module DocMyRoutes
+  module HashHelpers
+    def self.deep_merge(first, second)
+      merger = proc { |key, f, s| f.is_a?(Hash) && s.is_a?(Hash) ? f.merge(s, &merger) : s }
```
Comment (punkle) [inline lib/doc_my_routes/doc/hash_helpers.rb]: Can we name these variables oldval, newval or something other than single chars?

File: lib/doc_my_routes/doc/route.rb [hunk 3]
```diff
@@ -23,11 +24,11 @@ def initialize(resource, verb, route_pattern, conditions, documentation)
     end
 
     def to_hash
-      {
+      HashHelpers.deep_merge({
```
Comment (punkle) [inline lib/doc_my_routes/doc/route.rb]: Can you add a comment explain why deep merge is necessary.

Comment (vasyl-purchel) [reply lib/doc_my_routes/doc/route.rb]: done

File: lib/doc_my_routes/doc/hash_helpers.rb (hunk 2 above)
Comment (vasyl-purchel) [reply lib/doc_my_routes/doc/hash_helpers.rb]: done

File: lib/doc_my_routes/doc/hash_helpers.rb (hunk 1 above)
Comment (vasyl-purchel) [reply lib/doc_my_routes/doc/hash_helpers.rb]: done

**Comment by punkle** [comment]
Looks really good. Would you mind adding a short update to the README.md about how to use this parameter annotation.

**Pull request merged**